test/topology_custom/test_mv_fail_updates.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import asyncio
+import pytest
+import time
+from test.topology.conftest import skip_mode
+from test.pylib.manager_client import ManagerClient
+
+from cassandra.cluster import ConsistencyLevel  # type: ignore
+from cassandra.query import SimpleStatement  # type: ignore
+
+
+async def wait_for_views(cql, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        if done[0][0] == node_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+# This test makes sure that even if the view updates generation is failing on a node,
+# the reads can still succeed, as long as we can reach quorum.
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_mv_fail_updates(manager: ManagerClient) -> None:
+    node_count = 3
+    servers = await manager.servers_add(node_count)
+    cql = manager.get_cql()
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': 3}}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, PRIMARY KEY (key, c)) WITH SPECULATIVE_RETRY = 'ALWAYS'")
+    # Insert initial rows for building an index
+    for i in range(100):
+        await cql.run_async(f"INSERT INTO ks.tab (key, c) VALUES ({i}, 0)")
+
+    # The exception type used for the test is the one that is thrown when a replica write times out
+    # during the view update generation, but any exception other than the timeout is handled the same way.
+    await manager.api.enable_injection(servers[2].ip_addr, 'timeout_view_update_generation', one_shot=False)
+    await cql.run_async(f"CREATE INDEX tab_by_c ON ks.tab (c)")
+    # Insert more rows while building an index
+    for i in range(100, 200):
+        await cql.run_async(f"INSERT INTO ks.tab (key, c) VALUES ({i}, 0)")
+    await wait_for_views(cql, node_count)
+
+    # Read using the index
+    with pytest.raises(Exception):
+        # cl ALL can't finish if replica writes failed and the read repair fails as well
```

Comment:
I understand what this test does, but don't understand what it wants to check:
Ideally, a CL=ALL should work correctly if one of the nodes is missing the data - but it waits until after the read-repair finishes. If I understand correctly, in your case read-repair writes to the one node with the missing data, and that does a view update and fails - so the read repair fails, and if the read repair fails, the original read fails.
Is this really the intended behavior of read repair? Doesn't it make more sense to **ignore** a failed read repair and not break the read?
But more importantly, I don't understand - is the goal of this test to test what read repair does? Why?

BTW, I would have loved to see this test split up into multiple smaller tests that test how smaller specific things work. E.g., if here you want to test read repair, you don't need "view building" and 200 rows and the rest of the stuff, just one row can show this problem. But  if you do split this test, it should use a fixture (or some other technique) to reuse the same cluster.